sdk/libs-scala/crypto/src/test/scala/com/daml/crypto/MessageSignaturePrototypeSpec.scala

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.crypto
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.scalatest.BeforeAndAfterAll
+
+import java.security.{KeyPairGenerator, SecureRandom, Security}
+import java.security.spec.ECGenParameterSpec
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class MessageSignaturePrototypeSpec extends AnyFlatSpec with Matchers with BeforeAndAfterAll {
+  behavior of MessageSignaturePrototype.getClass.getSimpleName
+
+  override def beforeAll(): Unit = {
+    val _ = Security.insertProviderAt(new BouncyCastleProvider, 1)
```

Comment:
In order for `Secp256k1` curve to be available, we appear to need to force BouncyCastle to be at the start of the chain

Reply:
When I originally did that, I hit issues when I tried to load the `Secp256k1` curve (e.b. with `new ECGenParameterSpec("secp256k1")`). IIRC, the typical exception message was along the lines of "curve not found".

Placing BC at the start of the provider chain side stepped this issue.

Maybe there's some additional configuration that I need to do here?

Reply:
Specific exception at test time is:
```
- should be able to sign and verify messages *** FAILED *** (117 milliseconds)
  java.security.SignatureException: Curve not supported: org.bouncycastle.jce.spec.ECNamedCurveSpec@64a8c844
  at jdk.crypto.ec/sun.security.ec.ECDSASignature.engineSign(ECDSASignature.java:486)
  at java.base/java.security.Signature$Delegate.engineSign(Signature.java:1423)
  at java.base/java.security.Signature.sign(Signature.java:712)
  at com.daml.crypto.MessageSignaturePrototypeUtil.sign(MessageSignaturePrototypeUtil.scala:15)
  at com.daml.crypto.MessageSignaturePrototypeSpec.$anonfun$new$2(MessageSignaturePrototypeSpec.scala:35)
  ...
```

Reply:
even when you do `KeyPairGenerator.getInstance("EC", "BC")` to make sure the engine for key gen is the one from BC?

Reply:
OK, resolved the my issue above. When fetching the signatures, its necessary to specify the provider - so use: `Signature.getInstance(algorithm, "BC")`